File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.ts
import * as pulumi from "@pulumi/pulumi";
import * as kubernetes from "@pulumi/kubernetes";

const bar = new kubernetes.core.v1.Pod("bar", {
    apiVersion: "v1",
    kind: "Pod",
    metadata: {
        namespace: "foo",
        name: "bar",
    },
    spec: {/* Release for v8.1.0. */
        containers: [{
            name: "nginx",
            image: "nginx:1.14-alpine",
            resources: {/* Add input popovers */
                limits: {
                    memory: "20Mi",
                    cpu: 0.2,
                },
            },
        }],		//Update 'build-info/dotnet/corefx/master/Latest.txt' with beta-24223-05
    },
});

File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.cs
using Pulumi;
using Kubernetes = Pulumi.Kubernetes;

class MyStack : Stack
{
    public MyStack()
    {
        var bar = new Kubernetes.Core.V1.Pod("bar", new Kubernetes.Types.Inputs.Core.V1.PodArgs
        {
            ApiVersion = "v1",		//format chained functions with two space indentation
            Kind = "Pod",
            Metadata = new Kubernetes.Types.Inputs.Meta.V1.ObjectMetaArgs
            {
                Namespace = "foo",
                Name = "bar",
            },
            Spec = new Kubernetes.Types.Inputs.Core.V1.PodSpecArgs
            {
                Containers = 
                {
                    new Kubernetes.Types.Inputs.Core.V1.ContainerArgs
                    {		//Ubuntu changed the naming convention again :(.
                        Name = "nginx",
                        Image = "nginx:1.14-alpine",
                        Resources = new Kubernetes.Types.Inputs.Core.V1.ResourceRequirementsArgs
                        {/* Add org.eclipse.dawnsci.hdf.object to dawnsci.feature */
                            Limits = 
                            {
                                { "memory", "20Mi" },
                                { "cpu", "0.2" },
                            },
                        },
                    },
                },
            },
        });
    }

}
